### PR TITLE
Chore: Refactor Pull Payment

### LINKED ISSUE #59: Refactor pullPayment contract to allow upgradability
The current construct of the pullPayment satellite doesn't allow Upgrade-ability.

The storage inside the PullPayment satellite disallows the escrow council to upgrade the PullPayment contract in case of any issue.

Following changes need to be made
- [ ] Refactor logic for PullPayment in the `MarketEnabled`of  controller.

- [ ] Put all storage in the storage satellite

- [x] Add self destruct in the PullPayment in case of emergency (Storage still being in a separate contract allows us to reintegrate a new PullPayment at will)

On cron vs async send from PullPayments,
cron job could get costly + slow in the future when we have more users withdrawing money. 

Pull by users is always better.

### PR DESCRIPTION
**Pull request by bkcninja**
Resolves #59 
Changes made-:
1. Moved all PullPayment data into storage
2. Moved all controller logic into MarketEnabled
3. Changed UpgradeEvent to incorporate the controller changes
4. Added self destruct in PullPayment to recover stolen ether in case of emergency.
5. Cleanup

Made small commits for simplicity in understanding. 
 
Note: None of the functions in the PullPayment satellite have changed names. The frontend can still call the same functions.

### CODE REVIEW

File: contracts/satelites/PullPayment.sol [hunk 1]
```diff
@@ -40,7 +44,7 @@ contract PullPayment is Ownable {
     assert(untrustedRecipient.call.gas(1000).value(amountWei)());
   }
 
-  function () public payable {
+  function () public payable onlyNutz {
```
Comment (bkcninja) [inline contracts/satelites/PullPayment.sol]: `onlyNutz` so that ether doesnt get locked in pullpayments by mistake

**Comment by bkcninja** [comment]
TEST CASE in upgradeEventCompact required-
1. The current design with monolithic PullPayment(current state in production) has to be upgraded to the newer design(refactored).

**Review by troggy — APPROVED** [review]
Do we need new tests for these changes? Otherwise looks good to me.

**Comment by johba37** [comment]
I don't understand yet. my confusion is this:

to react to a problem in pull payment (theft in progress), we need to be able to get to the ether. we can do this by either:
- kill pull payments
- change allocation in the storage

now we implement the ability to do both. i would have thought that just adding kill would have been enough.

I see 2 additional disadvantages moving the pullpayment logic to controller:
- any defect in controller can now compromise pullpayment
- transactions in pull payment become more expensive

why not simply add kill to pullPayment, but leave it a totally separate contract otherwise?

**Comment by bkcninja** [comment]
If we just add kill to it and dont refactor the storage into the storage satellite, then we wont be able to deploy and reintegrate a new pull payment.

**Comment by johba37** [comment]
because withdrawals should not cary a big amount of data (if i requested withdrawal, i most likely also executed withdrawal), i don't see a reason why we want to keep the storage between different contracts.

**Comment by troggy** [comment]
@johannbarbie what if we want to change daily limits?

**Comment by bkcninja** [comment]
Changed to #69

**Pull request closed without merging**